File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    
    for a in s:
        cnt = 0
        tmp = ord(a)
        while cnt < index:
            tmp += 1
            if tmp > 122:
                tmp = 97
            
            if chr(tmp) not in skip:
                cnt += 1

        answer += chr(tmp)
        
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    
    function changeIdx(codeAt) {
        let idx = 0;
        while (idx < index) {
            codeAt++
            if (!skip.includes(codeAt)) {
                idx++
            }
        }
        if (codeAt > 122) {
            return codeAt - 26
        }
        return codeAt
    }
    
    var answer = '';
    
    skip = skip.split('').map(char => char.charCodeAt());
    
    for (let char of s) {
        let codeAt = char.charCodeAt()
        answer += String.fromCharCode(changeIdx(codeAt))
    }  
    
    return answer;
}